### DISCUSSION TITLE
Question: Read multiple files as a single graph

### DISCUSSION THREAD

**Discussion by Ishitori** [networkit/networkit#1044] [Q&A] [answered]
I have a huge EdgeList pair graph it tsv format. It consists of multiple files and they are almost 1TB in size. I cannot really merge them into a single file.

Can I instruct Networkit to read a directory, which contains edgelist in tsv files, as a single graph?

**Comment by fabratu** [comment]
Hi, thanks for reaching out!

Unfortunately NetworKit does not directly support this for Edgelists. I am wondering though why merging isn't a valiable option. Is there a certain encoding used for the nodes and edges in the TSV-files? If not, it should be possible to concatenate them rather easily (given enough storage is available).

Another way would be to use data-reader like `pandas` to read each TSV-file and either save this as a "normal" edgelist or use `addNodes` and an `addEdge`-loop to add all nodes/edges to an empty graph. You can also do this one file after the other in order to minimize the overhead in memory allocation. After that you can also write the result as a `NetworKitBinaryGraph`, which will speed-up any future io-operation (especially useful for bigger graphs).

**Comment by fabratu** [comment, marked as answer]
Update:
Another way might be to actually create a graph for each file and merge them via `networkit.graphtools.merge(G1, G2)`. You only have to make care that the node IDs are consistent between different graphs (which depends on the actual data). My recommendation would be to set the EdgelistReader to treat the node IDs as `continous` (which is the default case).  Note, that this does likely create a memory overhead since the files only contain a portion of the graph. After the import of a file, the resulting graph contains all IDs but not all edges.

Pseudocode:
```
G1 = readfile("file1.tsv")
for myfile in list(file2.tsv,...,fileN.tsv)
    G2 = readfile(myfile)
    networkit.graphtools.merge(G1,G2)
end for
```

**Comment by fabratu** [comment]
Converting this to a discussion, since it might be of aid for others.
